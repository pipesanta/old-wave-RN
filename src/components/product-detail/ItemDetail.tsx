import React from 'react'
import { Text, View, StyleSheet } from 'react-native';
import { ResellerCard } from './ResellerCard';

interface Props {
    title: string;
    content: string;
    reseller?: any;
}

export const ItemDetail = ({ title, content, reseller }: Props) => {
    return (
        <View testID="ItemDetail_view" style={{ marginHorizontal: 20, marginBottom: 15 }}>

            <View style={{ flex: 1, flexDirection: 'row', alignItems: 'center' }}>
                <View>
                    <Text style={styles.subTitleDetail}>{title}</Text>
                </View>
                <View style={{ width: 10 }}></View>
                <View style={styles.hotizontalLine} />
            </View>

            {
                reseller
                    ?
                    <ResellerCard reseller={reseller} />
                    :
                    <Text testID="reseller_text_id" key="123" style={{ fontSize: 16, color: '#5C5E64', paddingLeft: 10, marginBottom: 20, textAlign: "justify" }}>
                        {content}
                    </Text>
            }


        </View>
    )
}

const styles = StyleSheet.create({

    subTitleDetail: {
        fontSize: 20,
        color: '#5C5E64',
        opacity: 0.5,
        fontWeight: 'normal'
    },
    hotizontalLine: {
        flex: 1,
        height: 1,
        alignContent: "center",
        padding: 'auto',
        backgroundColor: '#5C5E64',
        opacity: 0.5
    }

});
