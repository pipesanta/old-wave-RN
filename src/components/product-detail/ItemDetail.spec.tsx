
import React from "react"
import renderer from "react-test-renderer"

import { ItemDetail } from "./ItemDetail";

jest.runAllTimers();



describe("<ItemDetail />", () => {

    const tree: any = renderer.create(<ItemDetail
        content="contenido del eso"
        title="titulo"
        key="12321.123"
        reseller={{ value: "reseller" }}
    />)


    it("has 2 child", (done) => {
        expect(tree.toJSON()?.children?.length).toBe(2);
        done();
    });

    it("verify type of <ItemDetail /> ", (done) => {
        expect(tree.toJSON()?.type).toBe("View");
        done();
    });

    it("verify style props of <ItemDetail /> ", (done) => {

        const expectedStyle = { marginHorizontal: 20, marginBottom: 15 };

        expect(tree.toJSON().props.style).toMatchObject(expectedStyle);

        done();
    });


    it('snapshot ----------', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });

    // it("<ItemDetail /> text content...", () => {

    // console.log({
    //     ROOT: tree.root
    // });

    // const mainView = tree.findAllByType({ testID: "reseller_text_id" });

    // console.log("-------------", {
    //     children: mainView
    // });

    // const resellerText = mainView.root.findByProps({ testID: 'reseller_text_id' }).props

    // console.log({
    //     resellerText
    // });


    // console.log(mainView.children);
    // expect(resellerText.children).toEqual("contenido del eso");
    // });


})