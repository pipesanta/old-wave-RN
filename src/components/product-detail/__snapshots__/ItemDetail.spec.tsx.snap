// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetail /> snapshot ---------- 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
      "marginHorizontal": 20,
    }
  }
  testID="ItemDetail_view"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#5C5E64",
            "fontSize": 20,
            "fontWeight": "normal",
            "opacity": 0.5,
          }
        }
      >
        titulo
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#5C5E64",
          "flex": 1,
          "height": 1,
          "opacity": 0.5,
          "padding": "auto",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 9,
        "flexDirection": "row",
        "height": 50,
        "marginLeft": 50,
        "marginRight": 50,
        "paddingRight": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
</View>
`;
